### PR TITLE
Add non-rollon DH to initialization action

### PR DESCRIPTION
**Pull request by nilsdeppe**
## Proposed changes

The final piece of the new damped wave gauge: initialization through the `InitializeDampedHarmonic` action. This also removes compute tags that are no longer used.

I'm going to submit one more PR that allows specifying the various gauge options in the input file.

New commits:
- Add non-rollon DH to InitializeDH action
- Remove unused compute tags from GH

Depends on:
- [x] #2339 

### Types of changes:

- [ ] Bugfix
- [ ] New feature
- [ ] Refactor

### Component:

- [ ] Code
- [ ] Documentation
- [ ] Build system
- [ ] Continuous integration

### Code review checklist

- [ ] The PR passes all checks, including unit tests and `clang-tidy`.
  For instructions on how to perform the CI checks locally refer to the [Dev
  guide on the Travis CI](https://spectre-code.org/travis_guide.html).
- [ ] The code is documented and the documentation renders correctly. Run
  `make doc` to generate the documentation locally into `BUILD_DIR/docs/html`.
  Then open `index.html`.
- [ ] The code follows the stylistic and code quality guidelines listed in the
  [code review guide](https://spectre-code.org/code_review_guide.html).

### CODE REVIEW

**Review by geoffrey4444 — DISMISSED** [review]
Looks good to me!

**Review by geoffrey4444 — DISMISSED** [review]

File: src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.cpp [hunk 1]
```diff
@@ -110,10 +111,87 @@ InitializeDampedHarmonic<Dim>::impl_rollon(
...
+
+  const auto ex_curv =
+      extrinsic_curvature(spacetime_unit_normal_vector, pi, phi);
+  const auto trace_ex_curv = trace(ex_curv, inverse_spatial_metric);
+  const auto christoffel_first = gr::christoffel_first_kind(d_spatial_metric);
+  const auto trace_christoffel_first =
+      trace_last_indices(christoffel_first, inverse_spatial_metric);
+
+  typename DampedHarmonicCompute<Frame::Inertial>::return_type
+      gauge_h_and_deriv{};
```
Comment (moxcodes) [inline src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.cpp]: Initialize with size if it's known here

File: src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.cpp [hunk 2]
```diff
@@ -110,10 +111,87 @@ InitializeDampedHarmonic<Dim>::impl_rollon(
...
+  DampedHarmonicCompute<Frame::Inertial>::function(
+      make_not_null(&gauge_h_and_deriv), lapse, shift,
+      spacetime_unit_normal_one_form, sqrt_det_spatial_metric,
+      inverse_spatial_metric, spacetime_metric, pi, phi, coords, sigma_r);
+  const auto& gauge_h =
+      get<Tags::GaugeH<Dim, Frame::Inertial>>(gauge_h_and_deriv);
+
+  // Compute lapse and shift time derivatives
+  Scalar<DataVector> dt_lapse{};
+  get(dt_lapse) =
```
Comment (moxcodes) [inline src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.cpp:174]: these calculations are a little surprising to me -- for instance, there's parts that I would have anticipated to be computed equivalently to `time_deriv_of_lapse`. I take it that the parts proportional to `gauge_h` components are additional sources associated with the gauge function, but in each of these calculations there's parts that appear independent of the gauge. Would it be incorrect to calculate those parts from the existing free functions?

File: src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.hpp [hunk 3]
```diff
@@ -149,41 +233,75 @@ struct InitializeDampedHarmonic {
...
+          db::get<::Initialization::Tags::InitialTime>(box);
+      const auto inertial_coords =
+          db::get<::domain::CoordinateMaps::Tags::CoordinateMap<
+              Metavariables::volume_dim, Frame::Grid, Frame::Inertial>>(box)(
+              db::get<::domain::Tags::ElementMap<Metavariables::volume_dim,
+                                                 Frame::Grid>>(box)(
+                  db::get<domain::Tags::Coordinates<Dim, Frame::Logical>>(box)),
+              initial_time, db::get<::domain::Tags::FunctionsOfTime>(box));
+
+      auto gauge_corrected_pi = new_pi_from_gauge_h(
```
Comment (moxcodes) [inline src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.hpp]: is it not better to make the `new_pi_from_gauge_h` function a return-by-pointer function and use it as the mutator for the `db::mutate` call?

Comment (nilsdeppe) [reply src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.hpp]: I personally dislike the "mutator" business, finding it rather difficult to read.

File: src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.cpp (hunk 2 above)
Comment (nilsdeppe) [reply src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.cpp:174]: I would have to go through the math in detail. This is implementing Eq 9 and 10 of https://arxiv.org/pdf/gr-qc/0512093.pdf directly, which I find might be clearer :shrug:

File: src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.hpp (hunk 3 above)
Comment (moxcodes) [reply src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.hpp]: I'm fine with leaving a lambda in the db::mutate call, but I do think it's better to perform the calculation in a return-by-pointer function; I think you can save an allocation/copy, right?

File: src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.cpp (hunk 2 above)
Comment (moxcodes) [reply src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.cpp:174]: Ah, fair enough -- if this is just directly implementing a published result, I won't ask for math alterations. Would you mind adding a citation to that paper in the dox?

**Comment by nilsdeppe** [comment]
Rebased and squashed. Thanks for the reviews! :)

**Review by moxcodes — APPROVED** [review]
LGTM

**Review by geoffrey4444 — APPROVED** [review]

File: src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.cpp [hunk 4]
```diff
@@ -147,16 +229,48 @@ void InitializeDampedHarmonic<Dim>::DampedHarmonicRollonCompute<Frame>::
...
+        const double sigma_r) noexcept {
+  if (UNLIKELY(h_and_d4_h->number_of_grid_points() != get(lapse).size())) {
+    h_and_d4_h->initialize(get(lapse).size());
+  }
+  damped_harmonic(
+      make_not_null(&get<Tags::GaugeH<Dim, Frame>>(*h_and_d4_h)),
+      make_not_null(&get<Tags::SpacetimeDerivGaugeH<Dim, Frame>>(*h_and_d4_h)),
+      lapse, shift, spacetime_unit_normal_one_form, sqrt_det_spatial_metric,
+      inverse_spatial_metric, spacetime_metric, pi, phi, coords, 1., 1.,
+      1.,       // amp_coef_{L1, L2, S}
```
Comment (kidder) [inline src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.cpp:255]: should/are these hard-coded values be documented somewhere?

Comment (nilsdeppe) [reply src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.cpp:255]: No they are not. I'm going to make them input file options in my next PR. They've been hard-coded for forever and I think it's time to change that :)

Comment (kidder) [reply src/Evolution/Systems/GeneralizedHarmonic/GaugeSourceFunctions/InitializeDampedHarmonic.cpp:255]: ok, as long as they are being changed in the next PR...

**Review by kidder — APPROVED** [review]

**Pull request merged**